tests/test_mixed_simulator_device.py

```diff
@@ -662,6 +701,58 @@ def test_expval_two_wires_with_parameters(
 
         assert np.isclose(res, expected_output, **tol)
 
+    def test_four_qubit_random_circuit(self, shots):
+        """Test a four-qubit random circuit with the whole set of possible gates"""
```

Comment:
Worth noting the connection between this test case and the one in the device test suite.